ACPs/77-reinventing-subnets/README.md

```diff
@@ -109,21 +76,77 @@ type SubnetValidator struct {
+
+The `validationID` for validators added in `ConvertSubnetTx` is defined as the SHA256 hash of `(convertSubnetTxID,validatorIndex)`, where `validatorIndex` refers to the index into the `Validators` array of within the `ConvertSubnetTx`.
+
+The following serialization is defined as an `InitialSubnetValidator`:
+```text
++--------------+----------+-----------+
+|       nodeID : [32]byte |  32 bytes |
++--------------+----------+-----------+
+|       weight :   uint64 |   8 bytes |
++--------------+----------+-----------+
+| blsPublicKey : [48]byte |  48 bytes |
++--------------+----------+-----------+
+                          |  88 bytes |
+                          +-----------+
+```
+The following serialization is defined as the `SubnetConversionData`
+```text
++-------------------+--------------------------+--------------------------------------------------------------+
+| convertSubnetTxID :                 [32]byte |                                                     32 bytes |
++-------------------+--------------------------+--------------------------------------------------------------+
+|    managerChainID :                 [32]byte |                                                     32 bytes |
++-------------------+--------------------------+--------------------------------------------------------------+
+|    managerAddress :                   []byte |                                4 + len(managerAddress) bytes |
++-------------------+--------------------------+--------------------------------------------------------------+
+| initialValidators : []InitialSubnetValidator |                        4 + len(initialValidators) * 88 bytes |
++-------------------+--------------------------+--------------------------------------------------------------+
+                                               | 72 + len(managerAddress) + len(initialValidators) * 88 bytes |
+                                               +--------------------------------------------------------------+
+```
+The `subnetConversionID` is defined as the SHA256 hash of the `SubnetConversionData` from a given `ConvertSubnetTx`.
+
+Once a `ConvertSubnetTx` is accepted, P-Chain validators will be willing to sign a `SubnetConversionMessage`, specified as an `AddressedCall` with a empty `originSenderAddress` with the following payload.
```

Comment:
It's unclear from the spec how `SubnetConversionData` is factored in here. How will Subnets consume `SubnetConversionData`?

Reply:
Validator manager contracts will need to learn of the validators specified in the `ConvertSubnetTx` in order to be able to remove them in the future. The `SubnetConversionData` includes all of those validators. It's expected for validator manager contracts to be provided the `SubnetConversionData` that is the pre-image of the `subnetConversionID` provided via a `SubnetConversionMessage`. 

We could have the `SubnetConversionData` itself be the `SubnetConversionMessage`, but it can be arbitrarily large due to the `validators` array, and the P-Chain will need to be willing to sign the `SubnetConversionMessage` in perpetuity for a given Subnet, so we should keep it to a fix size (similar to tracking Subnet IDs and Chain IDs on the P-Chain today).

Reply:
Got it- I was missing that nuance of the pre-image